pkg/cloud/api/mutable_resource.go

```diff
@@ -272,21 +302,38 @@ func (u *mutableResource[GA, Alpha, Beta]) AccessBeta(f func(x *Beta)) error {
 	return u.postAccess(meta.VersionBeta, 0)
 }
 
+// ImpliedVersion returns the implied version of the underlying resource.
+// This is determined by the convertibility of the resource.
+//
+// Note: CheckSchema validates that GA is not a PlaceholderType and that it is
+// convertible to Alpha and Beta.
+//
+//	y = Can convert to this version.
+//	n = Can't convert (note: nothing can convert to PlaceholderType)
+//
+// | GA| B | A |
+// -----------------
+// | y | y | y | GA
+// | n | y | y | Beta
+// | n | y | n | Beta
+// | n | n | y | Alpha
+// | n | n | n | Error
 func (u *mutableResource[GA, Alpha, Beta]) ImpliedVersion() (meta.Version, error) {
 	_, gaErr := u.ToGA()
-	_, alphaErr := u.ToAlpha()
+	if gaErr == nil {
```

Comment:
This code looks to be wrong?

Reply:
We are checking the conversion GA -> Beta and Alpha in CheckSum so I thought that there is no need to do this here because we enforce this is UT.
So I only check toGA and if this succeed I don't check toBeta and toAlpha because they can fail due to PlaceholderType and in CheckSum I checked that GA is a subset of B and A.

Reply:
I delete the table. It is incomplete because there are no information about PlaceholderType. Let me know what do you think.

Reply:
I added the table in comment